internal/docker/client.go

```diff
@@ -461,10 +461,16 @@ func (c *Cli) ImageBuild(ctx context.Context, buildContext io.Reader, options Bu
 		logger.Get(ctx).Verbosef("%v", c.initError)
 	}
 
+	// Always use a one-time session when using buildkit, since credential
+	// passing is fast and we want to get the latest creds.
+	// https://github.com/tilt-dev/tilt/issues/4043
 	var oneTimeSession *session.Session
 	sessionID := c.creds.sessionID
-	if len(options.SSHSpecs) > 0 || len(options.SecretSpecs) > 0 {
-		if c.builderVersion != types.BuilderBuildKit {
+
+	mustUseBuildkit := len(options.SSHSpecs) > 0 || len(options.SecretSpecs) > 0
+	isUsingBuildkit := c.builderVersion == types.BuilderBuildKit
+	if mustUseBuildkit || isUsingBuildkit {
+		if !isUsingBuildkit {
```

Comment:
nit: might be a bit clearer to structure as:

```
if isUsingBuildkit {
 // create session
} else if mustUseBuildkit {
 // return error
}
```

Reply:
i like that! done